lib/kitchen/driver/vagrant.rb

```diff
@@ -310,12 +310,20 @@ def run(cmd, options = {})
       # * `:log_subject` defaults to a String representation of the Driver's
       #   class name
       #
+      # Since vagrant does not support being run through bundler, we escape
+      # any bundler environment should we detect one.  Otherwise, subcommands
+      # will inherit our bundled environment.
+      # @see https://github.com/test-kitchen/kitchen-vagrant/issues/190
       # @see Kitchen::ShellOut#run_command
       def run_command(cmd, options = {})
         merged = {
           :use_sudo => config[:use_sudo], :log_subject => name
         }.merge(options)
-        super(cmd, merged)
+        if Object.const_defined?("Bundler")
+          Bundler.with_clean_env { super(cmd, merged) }
```

Comment:
@ksubrama do not feel bad. I have been misunderstood for like a billion years. It's super super sad :frowning:. 

Basically you'll likely need to do something like launch with a non-Ruby process (we used Go for this in Vagrant) which doesn't mangle the env, copy all keys in the current env to a different key, and then iterates and restores them before shelling out.